src/tlo/methods/healthsystem.py

```diff
@@ -2521,7 +2580,40 @@ def process_events_mode_2(self, hold_over: List[HSIEventQueueItem]) -> None:
                             # Recalculate call on officers based on squeeze factor.
                             for k in updated_call.keys():
                                 updated_call[k] = updated_call[k]/(squeeze_factor + 1.)
-
+                                
+                            # Recalculate call on officers including task shifting, which may result in
+                            # a change to required time
+                            if task_shifting_adopted:
+                                updated_call_inc_task_shift = {}
+                                # Go over all officers in updated_call
+                                for k in updated_call.keys():
+                                
+                                    # If task-shifting was requested for this officer, change name
+                                    # of officer and rescale original task by relevant factor
+                                    if k in task_shifting_adopted.keys():
+                                    
+                                        task_for_officer = updated_call[k]*task_shifting_adopted[k][1]
+                                        j = task_shifting_adopted[k][0]
+                                        
+                                        if j in updated_call_inc_task_shift.keys():
+                                        # If officer is already included in updated_call_inc_task_shift
+                                        # (e.g. because it was already performing own tasks as well as
+                                        # taking over that of officer not available) add to original task
+                                            updated_call_inc_task_shift[j] += task_for_officer
+                                        else:
+                                            updated_call_inc_task_shift[j] = task_for_officer
+                                    
+                                    # Else simply add original requirement to new call
+                                    else:
+                                        if k in updated_call_inc_task_shift.keys():
+                                            # Ensure that if this officer already present in dictionary
+                                            # this task is added, not overwritten
+                                            updated_call_inc_task_shift[k] += updated_call[k]
+                                        else:
+                                            updated_call_inc_task_shift[k] = updated_call[k]
+
+                                updated_call = updated_call_inc_task_shift
+                            
```

Comment:
As we already have this type of logic (when the ACTUAL_APPT_FOOTPRINT is different to the EXPECTED_APPT_FOOTPRINT), I wonder if this is duplicative.

Could we instead let the HSI event return an ACTUAL_APPT_FOOTPRINT that reflects the task-shifting, so that we re-use that logic?